server/doi/queries.ts

```diff
@@ -85,9 +88,16 @@ export const persistCrossrefDepositRecord = async (ids, depositJson) => {
 
 	const crossrefDepositRecord = await createCrossrefDepositRecord({ depositJson });
 
-	await targetModel.update({
-		crossrefDepositRecordId: crossrefDepositRecord.id,
-	});
+	// this is just to make typescript happy, update cannot be called on the union
+	if (targetModel instanceof Pub) {
+		await targetModel.update({
+			crossrefDepositRecordId: crossrefDepositRecord.id,
+		});
+	} else {
+		await targetModel.update({
+			crossrefDepositRecordId: crossrefDepositRecord.id,
+		});
+	}
```

Comment:
This happens once or twice more somewhere, because `update` uses the `this` type of the model you cannot call it on the union, very annoying.